src/fs/network/header.ts

```diff
@@ -87,12 +77,21 @@ export const put = async (
   })
   const links = link.arrToMap(linksArr)
   const cid = await basic.putLinks(links, key)
-  const pinsForHeader = linksArr.map(l => l.cid)
-  const pins = [
+
+  const pinsForHeader = linksArr.reduce((acc, cur) => ({
+    ...acc,
+    [`${cur.name}`]: cur.cid
+  }), {} as PinMap)
+
+  const pins = {
     ...pinsForHeader,
-    ...pinMapToList(header.pins),
-    cid
-  ]
+    index: {
+      ...header.pins,
+      '': pinsForHeader.index
+    },
+    '': cid
```

Comment:
What's this thing here with the empty string as object key?

Reply:
This is a map of all the CIDs needed to pin some objects so it might look like 
```ts
{
  mtime: "Qm...",
  size: "Qm...",
  index: { ... },
  "": "Qm..."
}
```
so the empty key just means its the CID for _that_ object. We could also just do a single slash (`/`)

_buuutt_ I think this stuff is going to be rewritten for the FLOOFS restructure that @expede and I are talking about: https://talk.fission.codes/t/floofs-restructuring/718/4